turbinia/workers/analysis/windows_acct.py

```diff
@@ -0,0 +1,142 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Task for analysing Windows account passwords."""
+
+import os
+import subprocess
+
+from turbinia import TurbiniaException
+from turbinia.evidence import EvidenceState as state
+from turbinia.evidence import ReportText
+from turbinia.lib import text_formatter as fmt
+from turbinia.lib.utils import bruteforce_password_hashes
+from turbinia.lib.utils import extract_artifacts
+from turbinia.workers import Priority
+from turbinia.workers import TurbiniaTask
+
+
+class WindowsAccountAnalysisTask(TurbiniaTask):
+  """Task to analyze Windows accounts."""
+
+  REQUIRED_STATES = [state.ATTACHED, state.DECOMPRESSED]
+
+  def run(self, evidence, result):
+    """Run the Windows Account worker.
+
+    Args:
+        evidence (Evidence object):  The evidence to process
+        result (TurbiniaTaskResult): The object to place task results into.
+    Returns:
+        TurbiniaTaskResult object.
+    """
+    # Where to store the resulting output file.
+    output_file_name = 'windows_account_analysis.txt'
+    output_file_path = os.path.join(self.output_dir, output_file_name)
+
+    # What type of evidence we should output.
+    output_evidence = ReportText(source_path=output_file_path)
+
+    try:
+      (location, num_files) = self._collect_windows_files(evidence)
+    except TurbiniaException as e:
+      result.close(
+          self, success=True,
+          status='No Windows account files found: {0:s}'.format(str(e)))
+      return result
+    if num_files < 2:
+      result.close(self, success=True, status='No Windows account files found')
+      return result
+    (creds, hashnames) = self._extract_windows_hashes(
+        result, os.path.join(location, 'Windows', 'System32', 'config'))
+    (report, priority, summary) = self._analyse_windows_creds(creds, hashnames)
+    output_evidence.text_data = report
+    result.report_priority = priority
+    result.report_data = report
+
+    # Write the report to the output file.
+    with open(output_file_path, 'wb') as fh:
+      fh.write(output_evidence.text_data.encode('utf-8'))
+
+    # Add the resulting evidence to the result object.
+    result.add_evidence(output_evidence, evidence.config)
+    result.close(self, success=True, status=summary)
+    return result
+
+  def _collect_windows_files(self, evidence):
+    try:
+      collected_artifacts = extract_artifacts(
+          artifact_names=['WindowsSystemRegistryFiles'],
+          disk_path=evidence.local_path, output_dir=self.output_dir)
+    except TurbiniaException as e:
+      raise TurbiniaException('artifact extraction failed: {}'.format(str(e)))
+
+    # Extract base dir from our list of collected artifacts
+    location = os.path.dirname(collected_artifacts[0])
+
+    return (location, len(collected_artifacts))
+
+  def _extract_windows_hashes(self, result, location):
```

Comment:
Even thought these are internal methods, could you add docstrings for these just for consistency?

Reply:
Done